app/routes/shop.py

```diff
@@ -114,4 +118,27 @@ def upload_stamp_image():
   response.raise_for_status()
   data = response.json()
 
-  return { 'data':  data }
\ No newline at end of file
+  return { 'data':  data }
+
+@bp.route('/upload', methods=['POST'])
+@jwt_required()
+def upload():
```

Comment:
i think that in these endpoints, we should also update the shop and user object, and add the url to those objects

Reply:
and then return the updated shop/user object as the response, instead of returning just the s3 url

Reply:
So in some cases, the object to be updated does not exist yet (because image upload happens before the dive shop is created) and there would be no way to update an object that does not yet exist.

Also, I think it helps to keep the the updating process rather consistent, and to only happen during the patch requests.

Reply:
We should switch the order and send the image after creating the shop, so that we have the shop ID or potentially modify this endpoint to do the entire creation process.

even in the other upload process I actually don’t like that we are uploading to s3 without making a DB entry at the same time, and would like to change it eventually.

Reply:
Got it.